test-requirements.txt

```diff
@@ -4,3 +4,4 @@ urllib3>=1.26.19,<2.0.0  # MIT
 pytest>=8.0.0,<9.0.0  # MIT License (MIT)
 pytest-timer>=1.0.0,<2.0.0  # MIT License (MIT)
 pytest-cov>=5.0.0,<7.0.0  # MIT License (MIT)
+parameterized>=0.9.0,<1.0.0  # BSD
```

Comment:
Minor question - why not use https://docs.pytest.org/en/stable/example/parametrize.html ?